STUDY.ConsoleApp.HabitLogger/STUDY.ConsoleApp.HabitLogger/CrudOperations.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Microsoft.Data.Sqlite;
+using STUDY.ConsoleApp.HabitLogger.Models;
+using System.Globalization;
+
+namespace STUDY.ConsoleApp.HabitLogger;
+
+internal class CrudOperations
+{
+    internal static void GetAllRecords()
+    {
+        Console.Clear();
+        using (var connection = new SqliteConnection(Program.connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText = "SELECT * FROM drinking_water ";
+
+            List<DrinkingWater> tableData = new();
+
+            SqliteDataReader tableDataReader = tableCmd.ExecuteReader();
+            if (tableDataReader.HasRows)
+            {
+                while (tableDataReader.Read())
+                {
+                    tableData.Add(new DrinkingWater
+                    {
+                        ID = tableDataReader.GetInt32(0),
+                        Date = DateTime.ParseExact(tableDataReader.GetString(1), "dd-MM-yy", new CultureInfo("en-US")),
+                        Quantity = tableDataReader.GetInt32(2)
+                    });
+                }
+            }
+            else Console.WriteLine("No rows found.");
+
+            connection.Close();
```

Comment:
Because of the using statement you don't have to close the connection. The using statement will take care of that.